tap_hubspot/__init__.py

```diff
@@ -537,6 +540,24 @@ def sync_owners():
 
     singer.write_state(STATE)
 
+
+def sync_engagements():
+    schema = load_schema("engagements")
+    singer.write_schema("engagements", schema, ["id"])
+
+    url = get_url("engagements_all")
+    params = {'limit': 250}
+    topLevelKey = "results"
+
+    for row in gen_request(url, params, topLevelKey, "hasMore", ["offset"], ["offset"]):
+        record = request(get_url("engagements_detail", engagement_id=row['engagement']['id'])).json()
+        record = xform(record, schema)
```

Comment:
The HubSpot docs and my own testing show that the data returned from the "engagements_all" endpoint matches that from the "engagements_detail" endpoint. Instead of grabbing the details and then xforming, you should just xform the row and remove the extra request.